### ISSUE TITLE
Can't find one or more curves

### ISSUE THREAD

**Issue by wvanderp** [PlotJuggler/PlotJuggler#883]
Thanks for contributing to PlotJuggler. You are great!

# Problem description

The style of prefixed XY curves can not be edited because they are not found.

```
Can't find one or more curves.
This message will be shown only once.
```

# Debug steps

I've already done some digging and found the following:

https://github.com/facontidavide/PlotJuggler/blob/a123c36350b4ef116c7d76a79c16832eec6f9643/plotjuggler_app/plotwidget.cpp#L810-L816

On these lines, we search for the data to generate the curve on the edit screen.

When the curve is `world/[siam,tre]`, then `curve_x = siam` and `curve_y = tre`, these will not be found in `_mapped_data.numeric` because all the values there have their full path. So, `world/siam`.

So, one solution could be to save `curve_x` and `curve_y` with their full path. 
Another is adding an attribute of the path to the `curve_element` and combining them on the fly.

I at first thought This would happen with all prefixed paths, but it seems to only happen with data loaded from CSV.

# Steps to reproduce (important)

download [csv_files.zip](https://github.com/facontidavide/PlotJuggler/files/13301586/csv_files.zip)
int there are two CSV files

- drag both of them into plotjuggler
- use the default suggested prefixes
- create an XY plot from the data in the A prefix
- the plot appears
- right click and select `Edit Curves...`
- an alarm is shown that says, `Can't find one or more curves`
- and then you can't change the setting of the curve

here is a video of me doing it:
https://github.com/facontidavide/PlotJuggler/assets/2423856/805bb81f-891c-412f-93ed-fddf52e18066

# Related bug

https://github.com/facontidavide/PlotJuggler/issues/756 Layout doesn't restore XY plots when multiple sources with prefixes are used
this bug seems related because it describes the same problem with the prefix and XY plots

**Comment by facontidavide** [comment]
PlotJuggler 4 stores and resolves XY curves by an opaque full catalog key (CurveDescriptor.name), not the stripped display name, so editing prefixed XY curves no longer fails to find them — this is exactly the fix suggested in the issue. Closing as fixed in PJ4.

---
*PlotJuggler 4 is now the official release and the default branch of this repo; PlotJuggler 3 is end-of-life and no longer receives features or fixes. If this still affects you in PJ4, please reopen or file a fresh issue — it will be looked at.*
